src/ILCompiler.CppCodeGen/src/CppCodeGen/DependencyNodeIterator.cs

```diff
@@ -53,7 +53,13 @@ private void AddTypeNode(EETypeNode node)
                 EETypeNode baseTypeNode;
                 if (node.Type.BaseType != null)
                 {
-                    _typeToNodeMap.TryGetValue(node.Type.BaseType, out baseTypeNode);
+                    TypeDesc baseType = node.Type.BaseType;
+                    if (_factory.Target.Abi == TargetAbi.CppCodegen &&
```

Comment:
`_factory.Target.Abi == TargetAbi.CppCodege` check should not be needed. This whole file is specific to CppCodeGen.

Reply:
Yes, thank you!